os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -582,10 +567,36 @@ void up_assert(const uint8_t *filename, int lineno)
 		asserted_location = (uint32_t)user_assert_location;
 		user_assert_location = 0x0;
 	} else {
-		asserted_location = (uint32_t)kernel_assert_location;
+		ARCH_GET_RET_ADDRESS(asserted_location)
+	}
+
+#ifdef CONFIG_SMP
```

Comment:
How about leaving some comment to explain below?

Reply:
Done. Thanks.